pkg/schemas/generator.go

```diff
@@ -105,11 +137,125 @@ func (g Generator) Generate(ctx *genall.GenerationContext) error {
 			documents[documentName] = document
 		}
 		document.Definitions[context.definitionNameFor(documentName, typeIdent)] = typeSchema
+
+		// Generate a schema for types with "fybrik:validation:object" marker
+		info, knownInfo := parser.Types[typeIdent]
+		if knownInfo {
+			if info.Markers.Get(objectMarker.Name) != nil {
+				documentName := fmt.Sprintf("%s.json", info.Name)
+				document, exists := documents[documentName]
+				listFields, _ := context.getFields(typeIdent)
+				schemaPtr := parser.Schemata[typeIdent]
+				context.removeExtraProps(typeIdent, &schemaPtr, &listFields)
```

Comment:
You probably need to set `additionalProperties: true` for each object you remove properties from.

Reply:
By default any additional properties are allowed. Also, I tried to add a property that is not in the json schema and it worked without adding "additionalProperties: true".